src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -130,11 +119,7 @@ public ExerciseResource(ExerciseRepository exerciseRepository, ExerciseService e
         log.debug("REST request to delete Exercise : {}", id);
         Exercise exercise = exerciseService.findOneLoadParticipations(id);
         if (Optional.ofNullable(exercise).isPresent()) {
-            Course course = exercise.getCourse();
-            User user = userService.getUserWithGroupsAndAuthorities();
-            if (!authCheckService.isInstructorInCourse(course, user) && !authCheckService.isAdmin()) {
-                return forbidden();
-            }
+            if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) return forbidden();
```

Comment:
Please be more careful when changing existing code that works. This should only be possible for instructors, so change it to `isAtLeastInstructorForExercise`

Reply:
Thanks for noticing, I'll be more careful!